lvmd/command/lvm.go

```diff
@@ -644,16 +644,102 @@ func (l *LogicalVolume) Snapshot(name string, cowSize uint64) (*LogicalVolume, e
 
 	var lvcreateArgs []string
 	if _, err := os.Stat("/run/systemd/system"); err != nil {
-		lvcreateArgs = []string{"-s", "-n", name, l.fullname}
+		lvcreateArgs = []string{"--type", "thin", "-s", "-n", name, l.fullname}
 	} else {
-		lvcreateArgs = []string{"-s", "-k", "n", "-n", name, l.fullname}
+		lvcreateArgs = []string{"--type", "thin", "-s", "-k", "n", "-n", name, l.fullname}
 	}
 	if err := CallLVM("lvcreate", lvcreateArgs...); err != nil {
 		return nil, err
 	}
 	return l.vg.FindVolume(name)
 }
 
+// Activate activates the logical volume for desired access.
+func (l *LogicalVolume) Activate(access string) error {
```

Comment:
The reason for naming the function `Activate` was because it was based on `activation` of logical volumes.
See https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_logical_volumes/assembly_lvm-activation-configuring-and-managing-logical-volumes

And `ChangeAttribute` somewhat refers to `lvchange`, which provides a lot more functionalities than just activation.

Let me know if the name still needs a change, will update it based on your suggestion :+1:

Reply:
Has this comment been addressed?

Reply:
Activate function is performing the intended functionality, I don't think we need to break that function down further. Wdyt?

Reply:
I would put this issue on hold, as one command can have multiple meanings and be difficult to name. If a good name idea comes along later, it is easy to modify.